### PR TITLE
Reintroduce PMacc Include

### PR DESCRIPTION
**Pull request by ax3l**
Follow-up to #1449, thanks to @BenjaminW3 for spotting my incorrect merge.

### CODE REVIEW

**Comment by erikzenker** [comment]
Well, sometimes it is hard to see were some type is coming from. We can just hope that our IDEs will warn us in the future. Otherwise, exhausting automatic header checks can help.

**Comment by ax3l** [comment]
in principle yes, but in this case it was obvious and the only include ^^

**Comment by BenjaminW3** [comment]
alpaka has such self-contained header checks in the CI. However, they are slow as each header has to be built by a single `.cpp` file (especially when using the nvcc compiler).

File: src/libPMacc/include/nvidia/functors/Assign.hpp [hunk 1]
```diff
@@ -22,6 +22,8 @@
 
 #pragma once
 
+#include "pmacc_types.hpp"
+
 namespace PMacc
 {
     namespace nvidia
```
Comment (PrometheusPi) [inline src/libPMacc/include/nvidia/functors/Assign.hpp]: Following the [coding guideline](https://github.com/ComputationalRadiationPhysics/picongpu/wiki/Coding-Guide-Lines#1-spaces-and-indentation) there should be no indentation here (and all following namespaces).

**Comment by PrometheusPi** [comment]
@ax3l Your name is missing in the license header.

**Comment by ax3l** [comment]
no it is not, I just reintroduced a fix I merged accidentally.

also, it's do not consider such changes to be above the required "threshold of originality" for future reference.

**Comment by PrometheusPi** [comment]
@ax3l That's fine withe me as long as `git blame` still works :sunglasses:

File: src/libPMacc/include/nvidia/functors/Assign.hpp (hunk 1 above)
Comment (ax3l) [inline src/libPMacc/include/nvidia/functors/Assign.hpp]: yes, but this PR is, as stated in the description, a bug fix and tries to fix exactly *one* thing.

all files inside `src/libPMacc/include/nvidia/functors` have wrong formatting, we are not picking things we come along in our PRs but we change focused and topic-based.

Comment (PrometheusPi) [inline src/libPMacc/include/nvidia/functors/Assign.hpp]: I agree - we should keep the keep the wrong indentation in mind (I will fix them, as soon as this pull request is merged).

Comment (ax3l) [inline src/libPMacc/include/nvidia/functors/Assign.hpp]: that's not really necessary but if you do so, pls keep #755 in mind

**Pull request merged**